packages/api-fetch/src/utils/response.js

```diff
@@ -9,20 +9,24 @@ import { __ } from '@wordpress/i18n';
  * @param {Response} response
  * @param {boolean}  shouldParseResponse
  *
- * @return {Promise} Parsed response
+ * @return {Promise<Response | any | null>} Parsed response.
  */
 const parseResponse = ( response, shouldParseResponse = true ) => {
 	if ( shouldParseResponse ) {
 		if ( response.status === 204 ) {
-			return null;
+			return Promise.resolve( null );
 		}
 
 		return response.json ? response.json() : Promise.reject( response );
 	}
 
-	return response;
+	return Promise.resolve( response );
```

Comment:
Why do we need to wrap this with `Promise.resolve()`? Seems like we already invoke a `Promise.resolve()` over `parseResponse()` in `parseResponseAndNormalizeError()` below?

Reply:
We do that just to normalize the return type of the function so that it's just a `Promise<any>` rather than a `Promise<any> | Response`. Maybe that normalization isn't necessary though?

Reply:
I'd be in favor of keeping the existing behavior, even if that makes typing it a little vaguer. Any improvements there could be left for future PRs. What do you think?

Reply:
I'll try it out and see if I can make it work without other runtime changes 😀

Reply:
Good catch, it turns out I was prematurely optimizing the return type when it wasn't necessary given it's usage. Thanks! Update pushed 😁